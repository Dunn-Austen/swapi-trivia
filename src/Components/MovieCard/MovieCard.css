.movie-card {
  background-color: orange;
  border: 2px solid black;
  display: flex;
  flex-direction: column;
  height: 150px;
  margin: 10px;
  padding: 20px;
  text-align: center;
  width: 200px;
}
