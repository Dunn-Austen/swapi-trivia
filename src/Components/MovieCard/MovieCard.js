import React from 'react';
import './MovieCard.css';

const MovieCard = () => {

  return (
    <article className="movie-card">
      <h2 className="character-name">Movie Title</h3>
      // <img className="movie-img" src={} alt={}/>
      // Commented out so that empty attribute fields don't cause bugs until we eventually pass props down into this component
      <div className="movie-details">
        <p>Placeholder for movie details passed from eventual fetch data</p>
        <p>To be reformatted / styled later</p>
        <p>Placeholder</p>
        <p>Placeholder</p>
      </div>
      <button id="we'll likely have to pass a specific property value here from our fetch data, example: {props.id?}" onClick={() => }>View Characters</button>
    </article>
  )
};

export default MovieCard;
