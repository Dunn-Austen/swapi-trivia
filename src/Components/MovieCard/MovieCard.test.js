import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from '../MovieCard/MovieCard';

describe('Card', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<MovieCard
      title="Bananas"
      number="B A N A N A S"
      releaseDate={3}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});