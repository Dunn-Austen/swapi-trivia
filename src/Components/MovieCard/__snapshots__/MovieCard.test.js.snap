// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot with all data passed in correctly 1`] = `
<article
  className="movie-card"
>
  <h2
    className="movie-name"
  >
    Bananas
  </h2>
  <h2
    className="movie-number"
  >
    B A N A N A S
  </h2>
  <h2
    className="movie-releaseDate"
  >
    3
  </h2>
  <button>
    View Characters
  </button>
</article>
`;
